frappe/public/js/frappe/form/controls/multiselect.js

```diff
@@ -13,7 +13,8 @@ frappe.ui.form.ControlMultiSelect = frappe.ui.form.ControlAutocomplete.extend({
 
 			replace: function(text) {
 				const before = this.input.value.match(/^.+,\s*|/)[0];
-				this.input.value = before + text + ", ";
+				const value = before + text;
+				this.input.value = (value.includes(" <") ? value.split(' <')[0] : value) + ", ";
```

Comment:
Multiselect is generic, it will not only be used for emails but for any arbitrary data. We can use something like how the link field does it. There we have a label and value. We can add anything to label, but only value will get appended.